colcon_ros/task/catkin/build.py

```diff
@@ -73,6 +73,19 @@ def add_arguments(self, *, parser):  # noqa: D102
                         'python2path', Path(args.install_base),
                         self.context.pkg.name,
                         'PYTHONPATH', str(rel_python_path), mode='prepend')
+
+        shell_extensions = get_shell_extensions()
+        file_extensions = []
+        for shell_extensions_same_prio in shell_extensions.values():
+            for shell_extension in shell_extensions_same_prio.values():
+                file_extensions += shell_extension.get_file_extensions()
+
+        # register custom hooks created via catkin_add_env_hooks
+        custom_hooks_path = Path(args.install_base) / \
+            'etc' / 'catkin' / 'profile.d'
+        for file_extension in sorted(file_extensions):
+            additional_hooks += custom_hooks_path.glob('*.' + file_extension)
```

Comment:
Please sort the added hooks before appending them to ensure that the order based on the filenames is maintained.

Reply:
Fixed